agr_api/src/main/java/org/alliancegenome/api/service/GeneService.java

```diff
@@ -58,20 +59,21 @@ public Gene getById(String id) {
         return ret;
     }
 
-    public JsonResultResponse<PhenotypeAnnotation> getPhenotypeAnnotations(String geneID, Pagination pagination) {
+    public JsonResultResponse<PhenotypeAnnotation> getPhenotypeAnnotations(List<String> geneIDs, Pagination pagination) {
         LocalDateTime startDate = LocalDateTime.now();
-        PaginationResult<PhenotypeAnnotation> list = phenoCacheRepo.getPhenotypeAnnotationList(geneID, pagination);
+        PaginationResult<PhenotypeAnnotation> list = phenoCacheRepo.getPhenotypeAnnotationList(geneIDs, pagination);
         JsonResultResponse<PhenotypeAnnotation> response = new JsonResultResponse<>();
         response.calculateRequestDuration(startDate);
         response.setResults(list.getResult());
         response.setTotal(list.getTotalNumber());
         return response;
     }
 
-    public EntitySummary getPhenotypeSummary(String geneID) {
+    public EntitySummary getPhenotypeSummary(List<String> geneIDs) {
+
         EntitySummary summary = new EntitySummary();
-        summary.setNumberOfAnnotations(phenoRepo.getTotalPhenotypeCount(geneID, new Pagination()));
-        summary.setNumberOfEntities(phenoRepo.getDistinctPhenotypeCount(geneID));
+        geneIDs.forEach(geneID ->summary.setNumberOfAnnotations(phenoRepo.getTotalPhenotypeCount(geneID, new Pagination())));
```

Comment:
It looks like this lambda loop is just overwriting each loop element's previous call, i.e. the setting of the number is set to various numbers where only the last call will be persisted.